src/main/java/cms/gov/madie/measure/services/VersionService.java

```diff
@@ -156,6 +169,11 @@ private void validateMeasureForVersioning(Measure measure, String username) {
           measure.getId());
       throw new BadVersionRequestException(
           "Measure", measure.getId(), username, "Measure has no CQL.");
+    } else {
+      final ElmJson elmJson = elmTranslatorClient.getElmJson(measure.getCql(), accessToken);
+      if (elmTranslatorClient.hasErrors(elmJson)) {
+        throw new CqlElmTranslationErrorException(measure.getMeasureName());
+      }
```

Comment:
We already have a boolean value **_cqlErrors_** to check any CQL Errors w.r.t to ELM translation. There is a check on line 157. Can you remind me why we may need to validate it again?

Reply:
@RohitKandimalla Thanks Rohit! That's because Praveena said she could still version a measure with back cql from the backend.

Reply:
Yep. The `cqlError` flag helps when communicating horizontally (UI <-> UI or service <-> service). During vertical calls (UI -> service) it's possible for the user to adjust the value.

Reply:
Yeah makes sense, this increases the total response time, but I guess we don't have any other way out for now.